src/primitive.ts

```diff
@@ -36,141 +36,167 @@ let LEADING_ZERO = /^[+-]?0[0-9_]/
+        // Multiline escape
+        ptr = skipVoid(str, ptr - 1, true)
+        if (str[ptr] !== '\n' && str[ptr] !== '\r') {
+          throw new TomlError('invalid escape: only line-ending whitespace may be escaped', {
+            toml: str,
+            ptr: tmp,
+          })
+        }
+        ptr = skipVoid(str, ptr)
+      } else if (c in ESC_MAP) {
+        // Classic escape
+        parsed += ESC_MAP[c as keyof typeof ESC_MAP]
+      } else {
+        throw new TomlError('unrecognized escape sequence', {
+          toml: str,
+          ptr: tmp,
+        })
+      }
+
+      sliceStart = ptr
+    } else if (!isLiteral && c === '\\') {
+      tmp = ptr - 1
+      isEscape = true
+      parsed += str.slice(sliceStart, tmp)
+    }
+  }
+
+  return parsed + str.slice(sliceStart, endPtr - 1)
 }
 
-export function parseValue (value: string, toml: string, ptr: number): boolean | number | TomlDate {
-	// Constant values
-	if (value === 'true') return true
-	if (value === 'false') return false
-	if (value === '-inf') return -Infinity
-	if (value === 'inf' || value === '+inf') return Infinity
-	if (value === 'nan' || value === '+nan' || value === '-nan') return NaN
-
-	if (value === '-0') return 0 // Avoid FP representation of -0
-
-	// Numbers
-	let isInt
-	if ((isInt = INT_REGEX.test(value)) || FLOAT_REGEX.test(value)) {
-		if (LEADING_ZERO.test(value)) {
-			throw new TomlError('leading zeroes are not allowed', {
-				toml: toml,
-				ptr: ptr
-			})
-		}
-
-		let numeric = +(value.replace(/_/g, ''))
-		if (isNaN(numeric)) {
-			throw new TomlError('invalid number', {
-				toml: toml,
-				ptr: ptr
-			})
-		}
-
-		if (isInt && !Number.isSafeInteger(numeric)) {
-			throw new TomlError('integer value cannot be represented losslessly', {
-				toml: toml,
-				ptr: ptr
-			})
-		}
-
-		return numeric
-	}
-
-	let date = new TomlDate(value)
-	if (!date.isValid()) {
-		throw new TomlError('invalid value', {
-			toml: toml,
-			ptr: ptr
-		})
-	}
-
-	return date
+export function parseValue(
+  value: string,
+  toml: string,
+  ptr: number,
+): boolean | number | TomlDate | BigInt {
+  // Constant values
+  if (value === 'true') return true
+  if (value === 'false') return false
+  if (value === '-inf') return -Infinity
+  if (value === 'inf' || value === '+inf') return Infinity
+  if (value === 'nan' || value === '+nan' || value === '-nan') return NaN
+
+  if (value === '-0') return 0 // Avoid FP representation of -0
+
+  // Numbers
+  let isInt
+  if ((isInt = INT_REGEX.test(value)) || FLOAT_REGEX.test(value)) {
+    if (LEADING_ZERO.test(value)) {
+      throw new TomlError('leading zeroes are not allowed', {
+        toml: toml,
+        ptr: ptr,
+      })
+    }
+
+    // Remove underscores for processing
+    const cleanValue = value.replace(/_/g, '')
+
+    // Check if it's an integer with more than 15 digits (exceeds safe integer precision)
+    if (isInt && /^[+-]?\d{16,}$/.test(cleanValue)) {
```

Comment:
Does this work for exponent notation such as `5e+22`? It looks like it's a string at this point, so I'm going to guess not.

There's also hex, octal, and binary: `0xdead`, `0b0110`, `0o777`. Not sure if that works as expected either.

Reply:
The expo notation is not an int value, according to the TOML spec (and the regex too)